src/main/java/org/jabref/gui/fieldeditors/KeywordsEditorViewModel.java

```diff
@@ -0,0 +1,90 @@
+import org.jabref.logic.integrity.FieldCheckers;
+import org.jabref.model.entry.Keyword;
+import org.jabref.model.entry.KeywordList;
+import org.jabref.model.entry.field.Field;
+import org.jabref.preferences.PreferencesService;
+
+public class KeywordsEditorViewModel extends AbstractEditorViewModel {
+
+    private final ListProperty<Keyword> keywordListProperty;
+    private final Character keywordSeparator;
+    private final SuggestionProvider<?> suggestionProvider;
+
+    public KeywordsEditorViewModel(Field field,
+                                   SuggestionProvider<?> suggestionProvider,
+                                   FieldCheckers fieldCheckers,
+                                   PreferencesService preferencesService,
+                                   UndoManager undoManager) {
+
+        super(field, suggestionProvider, fieldCheckers, undoManager);
+
+        keywordListProperty = new SimpleListProperty<>(FXCollections.observableArrayList());
+        this.keywordSeparator = preferencesService.getBibEntryPreferences().getKeywordSeparator();
+        this.suggestionProvider = suggestionProvider;
+
+        BindingsHelper.bindContentBidirectional(
+                keywordListProperty,
+                text,
+                this::serializeKeywords,
+                this::parseKeywords);
+    }
+
+    private String serializeKeywords(List<Keyword> keywords) {
+        return KeywordList.serialize(keywords, keywordSeparator);
+    }
+
+    private List<Keyword> parseKeywords(String newText) {
+        return KeywordList.parse(newText, keywordSeparator).stream().toList();
+    }
+
+    public ListProperty<Keyword> keywordListProperty() {
+        return keywordListProperty;
+    }
+
+    public StringConverter<Keyword> getStringConverter() {
+        return new StringConverter<>() {
+            @Override
+            public String toString(Keyword keyword) {
+                if (keyword == null) {
+                    return "";
+                }
+                return keyword.get();
+            }
+
+            @Override
+            public Keyword fromString(String keywordString) {
+                return new Keyword(keywordString);
+            }
+        };
+    }
+
+    public List<Keyword> getSuggestions(String request) {
+        List<Keyword> suggestions = suggestionProvider.getPossibleSuggestions().stream()
+                                                      .map(String.class::cast)
+                                                      .filter(keyword -> keyword.toLowerCase().contains(request.toLowerCase()))
+                                                      .map(Keyword::new)
+                                                      .distinct()
+                                                      .collect(Collectors.toList());
+
+        Keyword requestedKeyword = new Keyword(request);
+        if (!suggestions.contains(requestedKeyword)) {
+            suggestions.addFirst(requestedKeyword);
+        }
```

Comment:
This does not seem right. getSuggestions should return existent suggestions, not create new ones. This is not the job of this method. Should probably return an empty list, if there is no suggestion.

Reply:
The main reason I did this is to solve this problem. I want to add `Key` not `Key1` or `Key2`, so I can't do it until I add `Key1` and `Key2`.

![image](https://github.com/JabRef/jabref/assets/52158423/ee2e246e-9e2a-4ae5-ad64-6c0230064b75)

Reply:
Ah, ok. This makes sense. Does escape work?

Reply:
<kbd>ESC</kbd> works. Just tried.

Reply:
That seems to be a general javafx thing, there is another issue about it